examples/chat/client/tasks.nim

```diff
@@ -346,6 +460,16 @@ proc stopWakuChat2*() {.task(kind=no_rts, stoppable=false).} =
   if wakuState != WakuState.started: return
   wakuState = WakuState.stopping
 
+  if not wakuNode.wakuFilter.isNil():
+    let contentFilters = collect(newSeq):
+      for contentTopic in contentTopics:
+        ContentFilter(contentTopic: contentTopic)
+
+    await wakuNode.unsubscribe(FilterRequest(contentFilters: contentFilters,
+      pubSubTopic: DefaultTopic, subscribe: false))
+
+  wakuNode.unsubscribeAll(DefaultTopic)
+
```

Comment:
I'm curious why we unsubscribe when stopping waku? This is most likely due to my lack of understanding of waku.

Reply:
That's what nim-waku's own `chat2.nim` does; it may be that it helps clean up things both locally and the remote peer, I'm not completely certain.